September 2021 updates

shirolkar opened the pull request:
September 2021 updates, plus address all open issues: #45 #46 #48

fnaadb commented:
The Training failed on the amlcompute cluster, the agent node was my machine running WSL , here's the detailed error

Traceback (most recent call last):
  File "pipelines_master.py", line 123, in <module>
    data.download('.', show_progress=True)
  File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/graph.py", line 4680, in download
    show_progress=show_progress)
  File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 1597, in download
    show_progress=show_progress)
  File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 437, in download
    AzureFileDatastore._verify_prefix(prefix)
  File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 587, in _verify_prefix
    raise UserErrorException("Nested prefix '{}' for Azure File Share is currently not supported.")
UserScriptException: UserScriptException:
	Message: Nested prefix '{}' for Azure File Share is currently not supported.
	InnerException UserErrorException:
	Message: Nested prefix '{}' for Azure File Share is currently not supported.
	InnerException None
	ErrorResponse 
{
    "error": {
        "code": "UserError",
        "message": "Nested prefix '{}' for Azure File Share is currently not supported."
    }
}

shirolkar commented:
> The Training failed on the amlcompute cluster, the agent node was my machine running WSL , here's the detailed error
> 
> Traceback (most recent call last):
> File "pipelines_master.py", line 123, in
> data.download('.', show_progress=True)
> File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/graph.py", line 4680, in download
> show_progress=show_progress)
> File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 1597, in download
> show_progress=show_progress)
> File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 437, in download
> AzureFileDatastore._verify_prefix(prefix)
> File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 587, in _verify_prefix
> raise UserErrorException("Nested prefix '{}' for Azure File Share is currently not supported.")
> UserScriptException: UserScriptException:
> Message: Nested prefix '{}' for Azure File Share is currently not supported.
> InnerException UserErrorException:
> Message: Nested prefix '{}' for Azure File Share is currently not supported.
> InnerException None
> ErrorResponse
> {
> "error": {
> "code": "UserError",
> "message": "Nested prefix '{}' for Azure File Share is currently not supported."
> }
> }

@fnaadb Did you create a new ML WorkSpace or used an existing one? Its possible your default datastore is pointing to file store and not blob store. If you create a new ML WorkSpace the default points to blob store. Thanks. Jitendra

fnaadb commented:
> > The Training failed on the amlcompute cluster, the agent node was my machine running WSL , here's the detailed error
> > Traceback (most recent call last):
> > File "pipelines_master.py", line 123, in
> > data.download('.', show_progress=True)
> > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/graph.py", line 4680, in download
> > show_progress=show_progress)
> > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 1597, in download
> > show_progress=show_progress)
> > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 437, in download
> > AzureFileDatastore._verify_prefix(prefix)
> > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 587, in _verify_prefix
> > raise UserErrorException("Nested prefix '{}' for Azure File Share is currently not supported.")
> > UserScriptException: UserScriptException:
> > Message: Nested prefix '{}' for Azure File Share is currently not supported.
> > InnerException UserErrorException:
> > Message: Nested prefix '{}' for Azure File Share is currently not supported.
> > InnerException None
> > ErrorResponse
> > {
> > "error": {
> > "code": "UserError",
> > "message": "Nested prefix '{}' for Azure File Share is currently not supported."
> > }
> > }
> 
> @fnaadb Did you create a new ML WorkSpace or used an existing one? Its possible your default datastore is pointing to file store and not blob store. If you create a new ML WorkSpace the default points to blob store. Thanks. Jitendra

@shirolkar , After changing the default store from file to Blob store, it works, maybe we should make this part of the documentation.

fnaadb commented:
![image](https://user-images.githubusercontent.com/729486/132934704-e81c39dd-b55c-4999-a77b-0bcd00dccd5b.png)

Don't have the output directory after the training run successful

shirolkar commented:
@fnaadb In the latest update there is no output directory. Are you evaluating the MCW from my fork shirolkar:master or microsoft:master? The update is currently on my fork and I do not see a reference to output directory. Thanks. Jitendra

shirolkar commented:
> > > The Training failed on the amlcompute cluster, the agent node was my machine running WSL , here's the detailed error
> > > Traceback (most recent call last):
> > > File "pipelines_master.py", line 123, in
> > > data.download('.', show_progress=True)
> > > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/graph.py", line 4680, in download
> > > show_progress=show_progress)
> > > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/pipeline/core/_aeva_provider.py", line 1597, in download
> > > show_progress=show_progress)
> > > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 437, in download
> > > AzureFileDatastore._verify_prefix(prefix)
> > > File "/azureml-envs/azureml_7f7029d00755703cb49a1f2b0eb7936a/lib/python3.6/site-packages/azureml/data/azure_storage_datastore.py", line 587, in _verify_prefix
> > > raise UserErrorException("Nested prefix '{}' for Azure File Share is currently not supported.")
> > > UserScriptException: UserScriptException:
> > > Message: Nested prefix '{}' for Azure File Share is currently not supported.
> > > InnerException UserErrorException:
> > > Message: Nested prefix '{}' for Azure File Share is currently not supported.
> > > InnerException None
> > > ErrorResponse
> > > {
> > > "error": {
> > > "code": "UserError",
> > > "message": "Nested prefix '{}' for Azure File Share is currently not supported."
> > > }
> > > }
> > 
> > 
> > @fnaadb Did you create a new ML WorkSpace or used an existing one? Its possible your default datastore is pointing to file store and not blob store. If you create a new ML WorkSpace the default points to blob store. Thanks. Jitendra
> 
> @shirolkar , After changing the default store from file to Blob store, it works, maybe we should make this part of the documentation.

@fnaadb I will add a note in the setup where we create a new ML Workspace for case when people use an existing workspace.

fnaadb commented:
> @fnaadb In the latest update there is no output directory. Are you evaluating the MCW from my fork shirolkar:master or microsoft:master? The update is currently on my fork and I do not see a reference to output directory. Thanks. Jitendra

@shirolkar the MCW-ML-Ops workshop documentation shows the output directory, people might get confused

fnaadb commented:
![image](https://user-images.githubusercontent.com/729486/132937115-786e861c-e247-4097-9160-ccf332acff95.png)
the Test deployment fails , the image is from the test deployment release pipeline, fail to understand why the ML workspace will not have access to the storage account

shirolkar commented:
> ![image](https://user-images.githubusercontent.com/729486/132937115-786e861c-e247-4097-9160-ccf332acff95.png)
> the Test deployment fails , the image is from the test deployment release pipeline, fail to understand why the ML workspace will not have access to the storage account

@fnaadb Can you test with a newly created AML workspace (as recommend by the lab). You can also look at this link to see if it help to configure your datastore permissions https://docs.microsoft.com/en-us/azure/storage/blobs/assign-azure-role-data-access?tabs=portal. Thanks. Jitendra

p-lekkala commented:
- Exercise1: Task1 - step4 - "Select Notebooks on the left navigation pane, click 'go directly to my notebooks' "
- Exercise1: Task2 - while running "deep learning with text.ipynb" add  the following code
    run.wait_for_completion()
    For notebook to wait for completion of experiment (model building)

• Exercise3: Task2: 
		○ its not mentioned in pre-requisites that we need a linux VM, so its confusing where to run the instructions mentioned in the documentation
		○ Some parameters like "agent pool", "node pool" & "work folder" are not provided in the documentation

p-lekkala commented:
My jobs on build pipeline are not running, need help.
![image](https://user-images.githubusercontent.com/56669016/133460959-28c51e7f-a39d-4aa0-a7b5-0ed1de62243f.png)

shirolkar commented:
> * Exercise1: Task1 - step4 - "Select Notebooks on the left navigation pane, click 'go directly to my notebooks' "
> * Exercise1: Task2 - while running "deep learning with text.ipynb" add  the following code
>   run.wait_for_completion()
>   For notebook to wait for completion of experiment (model building)
> 
> • Exercise3: Task2:
> ○ its not mentioned in pre-requisites that we need a linux VM, so its confusing where to run the instructions mentioned in the documentation
> ○ Some parameters like "agent pool", "node pool" & "work folder" are not provided in the documentation
> ![image](https://user-images.githubusercontent.com/56669016/133460442-b6103ffd-1603-43d3-ba8a-6e9fbee9435b.png)
> ○ Provide a VM image or script if possible, for participants

@p-lekkala My updates in on my fork: https://github.com/shirolkar/MCW-ML-Ops. The PR is still open changes are not merged on the main branch. Look like you are following instructions on the main branch. Please test the instructions from my branch. Thanks. Jitendra

shirolkar commented:
> My jobs on build pipeline are not running, need help.
> ![image](https://user-images.githubusercontent.com/56669016/133460959-28c51e7f-a39d-4aa0-a7b5-0ed1de62243f.png)

@p-lekkala My updates in on my fork: https://github.com/shirolkar/MCW-ML-Ops. The PR is still open changes are not merged on the main branch. Look like you are following instructions on the main branch. Please test the instructions from my branch. In this case please make sure that your Agent is running. If you follow instructions from my fork, then there should not be an issue. Thanks. Jitendra

setuc commented:
I have added a pull request for shirolkar-master for new images. 

Some of these items are not clear

setuc commented:
Additional Feedback:
1. In the Before the HOL - MLOps.md, the Azure Agent can be a B1 machine. 
2. The AML Cluster does not need 2 machines, we can make do with one. Unless this is done in a corporate subscription, personal subscriptions *might* not have the quota to run this. Can we make this a low priority VM? Or reduce the idle seconds before scale down?
`az ml computetarget create amlcompute --min-nodes 0 --idle-seconds-before-scaledown 300 --vm-priority 'lowpriority'`

shirolkar commented:
> Additional Feedback:
> 
> 1. In the Before the HOL - MLOps.md, the Azure Agent can be a B1 machine.
> 2. The AML Cluster does not need 2 machines, we can make do with one. Unless this is done in a corporate subscription, personal subscriptions _might_ not have the quota to run this. Can we make this a low priority VM? Or reduce the idle seconds before scale down? Following suggestion for `azure-pipelines.yml`
>    `az ml computetarget create amlcompute --max-nodes 1 --min-nodes 0 --idle-seconds-before-scaledown 300 --vm-priority 'lowpriority'`

@setuc Hi Setu - we need 2 nodes. Since the master pipeline submits a AML pipeline. We can make a minor update in Q4 once we have v2 of the AML SDK that apparently have better support for running AML pipelines in Azure DevOps. Thanks. Jitendra

setuc commented:
The step to setup Azure CLI is different. We had set the scope limited to the subscription / workspace.

![image](https://user-images.githubusercontent.com/9305355/133567731-3388fbf9-7de0-4b94-a5a1-78f52e458e8b.png)

it actually should be like this:
![image](https://user-images.githubusercontent.com/9305355/133567875-92e50000-355b-4398-b429-0ea4e72ab454.png)

shirolkar commented:
> The step to setup Azure CLI is different. We had set the scope limited to the subscription / workspace.
> 
> ![image](https://user-images.githubusercontent.com/9305355/133567731-3388fbf9-7de0-4b94-a5a1-78f52e458e8b.png)
> 
> it actually should be like this:
> ![image](https://user-images.githubusercontent.com/9305355/133567875-92e50000-355b-4398-b429-0ea4e72ab454.png)

@setuc Hi Setu - I do not see a difference in the two screenshot. What am I missing? Thanks. Jitendra

setuc commented:
> @setuc Hi Setu - I do not see a difference in the two screenshot. What am I missing? Thanks. Jitendra

The ARM Connection -> This will be the subscription as not the workspace? It differs from the previous screenshot of your subscription. I dont think a subscription could possibly named quick-start-sc (technically you can, but not sure if the scope if different you in the screenshot).

shirolkar commented:
> > @setuc Hi Setu - I do not see a difference in the two screenshot. What am I missing? Thanks. Jitendra
> 
> The ARM Connection -> This will be the subscription as not the workspace? It differs from the previous screenshot of your subscription. I dont think a subscription could possibly named quick-start-sc (technically you can, but not sure if the scope if different you in the screenshot).

Its a service connection as created in Exercise 1 Task 4: Create new Service Connection. We use the same service connection in all the pipelines in the lab. The sc is scoped at ML Workspace. Thanks. Jitendra

shirolkar commented:
> I have added a pull request for shirolkar-master for new images.
> 
> Some of these items are not clear. For example, which Resource Group and Workspace are you referring to. Since we will be using to create the Azure Machine Pipeline, should we be updating the default resource group and ml workspace to the ones that we have suggested in the `Before the HOL - MLOps.md` file.
> ![image](https://user-images.githubusercontent.com/9305355/133563288-c07697cc-d7ff-4002-a22e-16ae70586460.png)
> 
> Suggest to update the
> 
> `variables: resourcegroup: 'MCW-MLOps' workspace: 'quick-start-ws' `

Hi @setuc Hi Setu - I have merged your PR and also updated the suggested image to show: `variables: resourcegroup: 'MCW-MLOps' workspace: 'quick-start-ws' ` Thanks. Jitendra

setuc commented:
Running the test deployment gives the following warnings:
`3 warnings
Resource file haven\'t set, can\'t find loc string for key: AmlCliNotAvailable
Resource file haven\'t set, can\'t find loc string for key: LoginFailed
Resource file haven\'t set, can\'t find loc string for key: ErrorInSettingUpSubscription`

I am unable to debug why that is the case. A similar issue was logged in MLOpsPython a while back; but seeing the same issue [here](https://github.com/microsoft/MLOpsPython/issues/120).

The step is successfully completed but with these 3 warnings.

setuc commented:
Notebook suggestion: Since you are suggesting for the users to make use of the Notebooks in AML instance, you can still the step
```python
#Provide the Subscription ID of your existing Azure subscription
subscription_id = "" # <- needs to be the subscription with the Quick-Starts resource group

#Provide values for the existing Resource Group 
resource_group = "MCW-MLOps-XXXXX" # <- replace XXXXX with your unique identifier

#Provide the Workspace Name and Azure Region of the Azure Machine Learning Workspace
workspace_name = "quick-starts-ws-XXXXX" # <- replace XXXXX with your unique identifier
workspace_region = "eastus" # <- region of your Quick-Starts resource group

# ws = Workspace.create(
#     name = workspace_name,
#     subscription_id = subscription_id,
#     resource_group = resource_group, 
#     location = workspace_region,
#     exist_ok = True)
```

and directly use
```python
from azureml.core import Workspace
ws = Workspace.from_config()```

shirolkar commented:
> Notebook suggestion: Since you are suggesting for the users to make use of the Notebooks in AML instance, you can still the step
> 
> ```python
> #Provide the Subscription ID of your existing Azure subscription
> subscription_id = "" # <- needs to be the subscription with the Quick-Starts resource group
> 
> #Provide values for the existing Resource Group 
> resource_group = "MCW-MLOps-XXXXX" # <- replace XXXXX with your unique identifier
> 
> #Provide the Workspace Name and Azure Region of the Azure Machine Learning Workspace
> workspace_name = "quick-starts-ws-XXXXX" # <- replace XXXXX with your unique identifier
> workspace_region = "eastus" # <- region of your Quick-Starts resource group
> 
> # ws = Workspace.create(
> #     name = workspace_name,
> #     subscription_id = subscription_id,
> #     resource_group = resource_group, 
> #     location = workspace_region,
> #     exist_ok = True)
> ```
> 
> and directly use
> 
> ```python
> from azureml.core import Workspace
> ws = Workspace.from_config()```
> ```

@setuc  Fixed the two notebooks that had this issue:  Model Telemetry.ipynb and Test Deployment.ipynb. Thanks.

setuc commented:
@shirolkar  Suggestion: I like the idea of test and approve, but one thought could be to introduce the test as a pipeline step itself. So, the pipeline step could check if everything is ok before the approval. Just a thought.

shirolkar commented:
> @shirolkar Suggestion: I like the idea of test and approve, but one thought could be to introduce the test as a pipeline step itself. So, the pipeline step could check if everything is ok before the approval. Just a thought.

setuc commented:
@shirolkar One more suggestion is that you can use Deploy To azure button as one click deployment to setup the initial steps to setup the RG and AML WS in Azure

p-lekkala commented:
@shirolkar 
Exercise2: task2:
MCW agent pool is not available to "mlops-quickstart" by default. This might be because Agent pool is created prior to project creation.
Pipeline run fails, it worked only after adding "MCW Agent pool" to the project.

shirolkar commented:
> @shirolkar
> Exercise2: task2:
> MCW agent pool is not available to "mlops-quickstart" by default. This might be because Agent pool is created prior to project creation.
> Pipeline run fails, it worked only after adding "MCW Agent pool" to the project.

@p-lekkala  Good to hear that its working now. In Task 4, Step 6 of **Before the HOL – MLOps**. It requires that you set the following setting on the `Agent pool: Auto-provision this agent pool in all projects: **Selected**`. This should work even if you create projects after creating the Agent. Thanks. Jitendra

p-lekkala commented:
@shirolkar 
Exercise3: Task2:
I'm getting the following error when adding an artifact.

Failed to query service connection API: 'https://centralindia.modelmanagement.azureml.net/api/subscriptions/28fff146-9fbe-44c3-9c39-d3edcfd806b3/resourceGroups/cyient/providers/Microsoft.MachineLearningServices/workspaces/aml-dev/models?count=50&api-version=2018-03-01-preview'. Error Message: 'An error occurred while sending the request.'

This error pops up when is select service Endpoint. How can I resolve it?

shirolkar commented:
> @shirolkar
> Exercise3: Task2:
> I'm getting the following error when adding an artifact.
> 
> Failed to query service connection API: 'https://centralindia.modelmanagement.azureml.net/api/subscriptions/28fff146-9fbe-44c3-9c39-d3edcfd806b3/resourceGroups/cyient/providers/Microsoft.MachineLearningServices/workspaces/aml-dev/models?count=50&api-version=2018-03-01-preview'. Error Message: 'An error occurred while sending the request.'
> 
> This error pops up when is select service Endpoint. How can I resolve it?

@p-lekkala This part of the lab has not changed in this update and has been in place for a while. You can try to recreate your service connection as per Exercise 1 / Task 4 make sure that Security: Grant access permission to all pipelines is checked. I am assuming you have created the service connection the Machine Learning Workspace scope level. Thanks. Jitendra

p-lekkala commented:
> > @shirolkar
> > Exercise3: Task2:
> > I'm getting the following error when adding an artifact.
> > Failed to query service connection API: 'https://centralindia.modelmanagement.azureml.net/api/subscriptions/28fff146-9fbe-44c3-9c39-d3edcfd806b3/resourceGroups/cyient/providers/Microsoft.MachineLearningServices/workspaces/aml-dev/models?count=50&api-version=2018-03-01-preview'. Error Message: 'An error occurred while sending the request.'
> > This error pops up when is select service Endpoint. How can I resolve it?
> 
> @p-lekkala This part of the lab has not changed in this update and has been in place for a while. You can try to recreate your service connection as per Exercise 1 / Task 4 make sure that Security: Grant access permission to all pipelines is checked. I am assuming you have created the service connection the Machine Learning Workspace scope level. Thanks. Jitendra

Recreated service connection again, but having the same issue. Is there anything else I can test?

shirolkar commented:
> > > @shirolkar
> > > Exercise3: Task2:
> > > I'm getting the following error when adding an artifact.
> > > Failed to query service connection API: 'https://centralindia.modelmanagement.azureml.net/api/subscriptions/28fff146-9fbe-44c3-9c39-d3edcfd806b3/resourceGroups/cyient/providers/Microsoft.MachineLearningServices/workspaces/aml-dev/models?count=50&api-version=2018-03-01-preview'. Error Message: 'An error occurred while sending the request.'
> > > This error pops up when is select service Endpoint. How can I resolve it?
> > 
> > 
> > @p-lekkala This part of the lab has not changed in this update and has been in place for a while. You can try to recreate your service connection as per Exercise 1 / Task 4 make sure that Security: Grant access permission to all pipelines is checked. I am assuming you have created the service connection the Machine Learning Workspace scope level. Thanks. Jitendra
> 
> Recreated service connection again, but having the same issue. Is there anything else I can test?

@p-lekkala Can you share screenshots / details log of the error? What artifact were you adding? Github repo or registered model? I am not sure if this is a permissions issue. Did you create the AML workspace as your user? Thanks. Jitendra

p-lekkala commented:
> > > > @shirolkar
> > > > Exercise3: Task2:
> > > > I'm getting the following error when adding an artifact.
> > > > Failed to query service connection API: 'https://centralindia.modelmanagement.azureml.net/api/subscriptions/28fff146-9fbe-44c3-9c39-d3edcfd806b3/resourceGroups/cyient/providers/Microsoft.MachineLearningServices/workspaces/aml-dev/models?count=50&api-version=2018-03-01-preview'. Error Message: 'An error occurred while sending the request.'
> > > > This error pops up when is select service Endpoint. How can I resolve it?
> > > 
> > > 
> > > @p-lekkala This part of the lab has not changed in this update and has been in place for a while. You can try to recreate your service connection as per Exercise 1 / Task 4 make sure that Security: Grant access permission to all pipelines is checked. I am assuming you have created the service connection the Machine Learning Workspace scope level. Thanks. Jitendra
> > 
> > 
> > Recreated service connection again, but having the same issue. Is there anything else I can test?
> 
> @p-lekkala Can you share screenshots / details log of the error? What artifact were you adding? Github repo or registered model? I am not sure if this is a permissions issue. Did you create the AML workspace as your user? Thanks. Jitendra

@shirolkar here is the screenshot
![image](https://user-images.githubusercontent.com/56669016/133617874-3425ccd6-184c-4485-842e-75b29e3dfa91.png)

shirolkar commented:
> > > > > @shirolkar
> > > > > Exercise3: Task2:
> > > > > I'm getting the following error when adding an artifact.
> > > > > Failed to query service connection API: 'https://centralindia.modelmanagement.azureml.net/api/subscriptions/28fff146-9fbe-44c3-9c39-d3edcfd806b3/resourceGroups/cyient/providers/Microsoft.MachineLearningServices/workspaces/aml-dev/models?count=50&api-version=2018-03-01-preview'. Error Message: 'An error occurred while sending the request.'
> > > > > This error pops up when is select service Endpoint. How can I resolve it?
> > > > 
> > > > 
> > > > @p-lekkala This part of the lab has not changed in this update and has been in place for a while. You can try to recreate your service connection as per Exercise 1 / Task 4 make sure that Security: Grant access permission to all pipelines is checked. I am assuming you have created the service connection the Machine Learning Workspace scope level. Thanks. Jitendra
> > > 
> > > 
> > > Recreated service connection again, but having the same issue. Is there anything else I can test?
> > 
> > 
> > @p-lekkala Can you share screenshots / details log of the error? What artifact were you adding? Github repo or registered model? I am not sure if this is a permissions issue. Did you create the AML workspace as your user? Thanks. Jitendra
> 
> @shirolkar here is the screenshot
> ![image](https://user-images.githubusercontent.com/56669016/133617874-3425ccd6-184c-4485-842e-75b29e3dfa91.png)

@p-lekkala Can you access the Models section in your AML Studio UI? Can you see any registered models in that UI?

shirolkar commented:
@p-lekkala Also, did you (your user) create a brand new AML Workspace for this lab? Or using an existing workspace? Thanks.

p-lekkala commented:
> @p-lekkala Also, did you (your user) create a brand new AML Workspace for this lab? Or using an existing workspace? Thanks.

Its an existing workspace, created about 2 months ago.

shirolkar commented:
@p-lekkala I recommend you create a new AML Workspace as described in before Before the HOL - MLOps.md and retest. Thanks. Jitendra

fnaadb commented:
![image](https://user-images.githubusercontent.com/729486/133634258-53ad51dc-5ba6-471c-a5ba-768c9570f132.png)
Even though i created a aks cluster, it doesn't show up in the drop down.
![image](https://user-images.githubusercontent.com/729486/133634435-4197d400-46e3-4869-8027-fecf1d0cee6b.png)

fnaadb commented:
![image](https://user-images.githubusercontent.com/729486/133659627-5d3a9871-73c4-4626-94c1-152fe468193a.png)
My production pipeline is failing, even though i have double checked the configuration
![image](https://user-images.githubusercontent.com/729486/133659801-9cd9a498-ff15-450f-a029-cbeeb8fe410d.png)

fnaadb commented:
Lab completed, except the optional features. No issues.

fnaadb commented:
> After turning on the appinsights, the only thing i don't see is the data in under the modeldata folder.
> 
> ![image](https://user-images.githubusercontent.com/729486/133667174-0549fb37-e496-43c7-8c6f-9c70a2ff81b9.png)
> 
> Other than that the lab is completed.

@shirolkar , please ignore the data showed up after a while

sdonohoo commented:
Review complete - no changes requested.

sureshpaulraj reviewed (changes requested):
Please update ML Workspace creation UI changes.

sureshpaulraj reviewed (changes requested):
ML Workspace creation UI changed

shirolkar commented on Hands-on lab/Before the HOL - MLOps.md:
```diff
@@ -31,6 +31,8 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
   - [Before the hands-on lab](#before-the-hands-on-lab)
     - [Task 1: Install the Microsoft DevLabs Machine Learning plugin for Azure DevOps](#task-1-install-the-microsoft-devlabs-machine-learning-plugin-for-azure-devops)
     - [Task 2: Create an Azure Machine Learning workspace](#task-2-create-an-azure-machine-learning-workspace)
```
@sureshpaulraj I have updated the screenshot and also added instructions for creating container registry. Thanks. Jitendra

shirolkar commented:
@danmanrique @timahenning All the actionable feedback from above is addressed. Thanks. Jitendra

DawnmarieDesJardins commented on Hands-on lab/HOL step-by-step - MLOps.md:
```diff
@@ -348,9 +226,9 @@ Duration: 45 minutes
     - Create the AML Compute target to run your master pipeline for model training and model evaluation.
 
-    - Run the master pipeline. The master pipeline has two steps: (1) Train the machine learning model, and (2) Evaluate the trained machine learning model. The evaluation step evaluates if the new model performance is better than the currently deployed model. If the new model performance is improved, the evaluate step will register the new model with Azure Machine Learning workspace. The results ...
+    - Run the master pipeline. The master pipeline has two steps: (1) Train the machine learning model, and (2) Evaluate the trained machine learning model. The evaluation step evaluates if the new model performance is better than the currently deployed model. If the new model performance is improved, the evaluate step will register the new model with Azure Machine Learning workspace. You can rev ...
```
Hello!  We cannot use the term master in documents.  Can the word be left out or changed to main?  (The image needs to be fixed too.)  It's also listed once in Exercise 3, task 2, instruction #3, under default branch.

shirolkar replied:
@DawnmarieDesJardins The default branch in Azure DevOps Git repo is called **master** and that is what Exercise 3, task 2 is referring to. Same applies to the image (trigger: -master). I cannot change those two instances. However, we can change the text in Exercise 2 Step 4 from master to main - there are 4 instances of the word master and we can change all 4 to main. I am assuming we can keep the file name pipelines_master.py (we should not change the filename). Please confirm if you want me to make the 4 instances of the word master to main in Exercise 2 Step 4. Thanks. Jitendra

shirolkar replied:
Also, we have used these names since the inception of the lab. So I am assuming this is a new restriction. Thanks.

shirolkar replied:
@DawnmarieDesJardins I am in IST for next two weeks, so our times may not overlap much. I went ahead and made changes to Exercise 2, task 4 to rename all instances of master to main. The only place where we refer to the word master will be in Exercise 3, task 2. Thanks. Jitendra

shirolkar replied:
@DawnmarieDesJardins  @timahenning I have removed all instances of the word **master** from the HOL step-by-step - MLOps.md markdown. Please review and let me know. Thanks.

DawnmarieDesJardins replied:
Much appreciated Jitendra!  Merging update now.
